### PR TITLE
DEVDOCS-3878: [maintenance] price list assignments

### PR DESCRIPTION
**Pull request by bc-andreadao**
# [DEVDOCS-3878]

Updated the price list assignment endpoints (e.g. PUT, POST, and DELETE price list assignments). I don't believe GET price list assignments needed updating (see ticket for screenshots).

[DEVDOCS-3878]: https://bigcommercecloud.atlassian.net/browse/DEVDOCS-3878?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: reference/price_lists.v3.yml [hunk 1]
```diff
@@ -2935,19 +3022,18 @@ components:
...
     PriceListAssignmentsIdentifiers:
       type: object
-      description: '`Price List Assignments` object used in GET response.'
       allOf:
         - type: object
           properties:
             customer_group_id:
               type: integer
-              description: The Customer Group with which this price list assignment is associated. A customer_group_id of 0 indicates default pricing for the channel given in the request.
+              description: The customer group with which this price list assignment is associated. A `customer_group_id` of 0 indicates default pricing for the channel given in the request.
```
Comment (bc-andreadao) [inline reference/price_lists.v3.yml:3030]: Should these be capitalized? I thought it looked over capitalized, but I also know that UI names should be in caps

File: reference/price_lists.v3.yml [hunk 2]
```diff
@@ -2935,19 +3022,18 @@ components:
...
             price_list_id:
               type: integer
-              description: The Price List with which this price set is associated.
+              description: The price list with which this price set is associated.
```
Comment (bc-andreadao) [inline reference/price_lists.v3.yml:3033]: Should these be capitalized? I thought it looked over capitalized, but I also know that UI names should be in caps

File: reference/price_lists.v3.yml [hunk 3]
```diff
@@ -2935,19 +3022,18 @@ components:
...
             channel_id:
               type: integer
-              description: The Channel with which this price set is associated. This field is optional or may be null when a price list applies across channels for the customer_group_id in the assignment.
+              description: The channel with which this price set is associated. This field is optional or may be `null` when a price list applies across channels for the `customer_group_id` in the assignment.
```
Comment (bc-andreadao) [inline reference/price_lists.v3.yml:3036]: Should these be capitalized? I thought it looked over capitalized, but I also know that UI names should be in caps

File: reference/price_lists.v3.yml (hunk 1 above)
Comment (bc-tgomez) [reply reference/price_lists.v3.yml:3030]: Doesn't need to be capitalized unless like you said, it's UI.

**Review by bc-tgomez — APPROVED** [review]
Looks great! Approving.

File: reference/price_lists.v3.yml (hunk 1 above)
Comment (bc-andreadao) [inline reference/price_lists.v3.yml:3030]:
```suggestion
              description: The Customer Group with which this price list assignment is associated. A `customer_group_id` of 0 indicates default pricing for the channel given in the request.
```

File: reference/price_lists.v3.yml (hunk 2 above)
Comment (bc-andreadao) [inline reference/price_lists.v3.yml:3033]:
```suggestion
              description: The Price List with which this price set is associated.
```

File: reference/price_lists.v3.yml (hunk 3 above)
Comment (bc-andreadao) [inline reference/price_lists.v3.yml:3036]:
```suggestion
              description: The Channel with which this price set is associated. This field is optional or may be `null` when a price list applies across channels for the `customer_group_id` in the assignment.
```

File: reference/price_lists.v3.yml (hunk 1 above)
Comment (bc-andreadao) [reply reference/price_lists.v3.yml]: k, I just found it in the UI, so I'll capitalize

File: reference/price_lists.v3.yml (hunk 2 above)
Comment (bc-andreadao) [reply reference/price_lists.v3.yml]: Just found it in the UI, so I'll capitalize

File: reference/price_lists.v3.yml (hunk 3 above)
Comment (bc-andreadao) [reply reference/price_lists.v3.yml]: Just found it in the UI, so I'll capitalize

File: reference/price_lists.v3.yml [hunk 4]
```diff
@@ -2681,7 +2706,7 @@ paths:
         - $ref: '#/components/parameters/ChannelIdInParam'
       responses:
         '204':
-          description: An empty response.
+          description: No Content.
```
Comment (markcmurphy) [inline reference/price_lists.v3.yml:2709]: Delete returns a `204 No Content`, but it does not return a body along with it. So the ref to `$/components/schemas/NoContent` can be removed.

File: reference/price_lists.v3.yml [hunk 5]
```diff
@@ -2668,6 +2668,31 @@ paths:
...
+        "422":
+          description: Error response. Includes the errors for each reference ID.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/PriceListAssignmentsBatchErrorResponse"
+    delete:
       tags:
         - Price Lists Assignments
       summary: Delete Price List Assignments
```
Comment (markcmurphy) [inline reference/price_lists.v3.yml:2698]: If no filter is selected, a `422` is returned. Can we indicate that at least one query param is required? 
```json
{
    "status": 422,
    "title": "At least one filter must be set for a delete action.",
    "type": "https://developer.bigcommerce.com/api-docs/getting-started/api-status-codes",
    "errors": {}
}
```

File: reference/price_lists.v3.yml [hunk 6]
```diff
@@ -2668,6 +2668,31 @@ paths:
...
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/SuccessBatchResponse"
+        "422":
+          description: Error response. Includes the errors for each reference ID.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/PriceListAssignmentsBatchErrorResponse"
```
Comment (markcmurphy) [inline reference/price_lists.v3.yml:2694]: This ref's schema doesn't match the format for the `422` I received:

```json
{
    "status": 422,
    "title": "Bulk Price List Assignment operation has failed",
    "type": "https://developer.bigcommerce.com/api-docs/getting-started/api-status-codes",
    "errors": {
        "0.price_list_id": "Request includes invalid price list id(s).",
        "0.customer_group_id": "The combination of customer_group_id 1 and channel_id 1 already assigned to price_list_id 4"
    },
    "meta": {
        "saved_records": 0
    }
}
```

File: reference/price_lists.v3.yml [hunk 7]
```diff
@@ -2692,6 +2717,31 @@ paths:
...
+            schema:
+              $ref: "#/components/schemas/AssignmentForPutRequest"
+        required: true
+      responses:
+        "200":
+          description: A price list assignment.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/AssignmentForPutResponse"
```
Comment (markcmurphy) [inline reference/price_lists.v3.yml:2741]: `PUT` `200` response appears to be missing the `data` and `meta` objects. This is the response I received:
```json
{
    "data": {
        "id": 11,
        "price_list_id": 4,
        "customer_group_id": 2,
        "channel_id": 1
    },
    "meta": {}
}
```
Within Stoplight, the response example is defaulting to negative integers. Can we either set a min or put in some example values? 
image

File: reference/price_lists.v3.yml [hunk 8]
```diff
@@ -2692,6 +2717,31 @@ paths:
...
+      operationId: upsertPriceListAssignment
+      parameters:
+        - $ref: "#/components/parameters/PriceListIdParam"
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/AssignmentForPutRequest"
+        required: true
+      responses:
```
Comment (markcmurphy) [inline reference/price_lists.v3.yml:2735]: Testing this with a `channel_id` of `0` or a negative number returns a `502` error:
```json
{
    "status": 502,
    "title": "Bad Gateway",
    "type": "https://developer.bigcommerce.com/api-docs/getting-started/api-status-codes",
    "errors": {}
}
``` 
We don't need to change anything, but you might want to report it as a bug to the engineering team.

File: reference/price_lists.v3.yml (hunk 5 above)
Comment (bc-andreadao) [reply reference/price_lists.v3.yml:2698]: kk, I added a sentence in the description for DELETE

File: reference/price_lists.v3.yml (hunk 4 above)
Comment (bc-andreadao) [reply reference/price_lists.v3.yml:2709]: kk, good catch, I removed the unneeded response body

File: reference/price_lists.v3.yml (hunk 6 above)
Comment (bc-andreadao) [reply reference/price_lists.v3.yml]: I changed it, see the screenshot. Not sure how detailed we can make it
422 error

File: reference/price_lists.v3.yml (hunk 7 above)
Comment (bc-andreadao) [reply reference/price_lists.v3.yml]: good catch, I updated the PUT 200 response and also added example values for successful PUT and GET responses

File: reference/price_lists.v3.yml (hunk 6 above)
Comment (markcmurphy) [reply reference/price_lists.v3.yml]: This looks good, thanks for updating

**Review by markcmurphy — APPROVED** [review]

**Pull request merged**